server.js

```diff
@@ -2,38 +2,55 @@ const app = require('express')();
 const server = require('http').createServer(app);
 const io = require('socket.io')(server);
 const Game = require('./Game/Game.js');
-let game;
+let game = new Game, players = [], generator;
 
 app.set('port', process.env.PORT || 3000);
 
-io.on('connection', (socket) => {
-  console.log('A user has connected.', io.engine.clientsCount)
-  io.sockets.emit(`User${io.engine.clientsCount} has connected`);
-  socket.on('disconnect', () => {
-    console.log('A user has disconnected.', io.engine.clientsCount)
-    io.sockets.emit(`User${io.engine.clientsCount} has disconnected`);
-  })
-})
-
 app.post('/', (req, resp) => {
-  if (req.act === 'join') {
-    joinGame(resp);
+  if (req.act === 'join' && typeof(req.role) === 'boolean') {
+    joinGame(req.role, resp);
   }
-  if (req.act === 'word') {
+  if (req.act === 'word' && req.word) {
     startGame(req, resp)
   }
   if (req.act === 'guess') {
     makeGuess(req, resp)
   }
-  resp.status(400).json("Incorrect 'act' verb");
+  resp.status(400).json("Incorrect 'act' verb or supplementary options");
 })
 
 app.get('/', (req, resp) => {
   resp.send('<h3>Connected</h3>');
 })
 
+joinGame(isGenerator, resp) {
+  let id = Date.now();
+  while (players.includes(id)) {
+    id -= 1;
```

Comment:
why the -= 1?

Reply:
To guarantee unique ID. Since multiple people could in theory try to access the game at the exact same millisecond, just added an easy way to keep them unique.